src/main/java/com/minecolonies/coremod/commands/citizencommands/ListCitizensCommand.java

```diff
@@ -59,37 +67,72 @@ public String getCommandUsage(@NotNull final ICommandSender sender)
         return super.getCommandUsage(sender) + "<colonyId>";
     }
 
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final List<ActionArgument> actionArgumentList,
+            @NotNull final Map<String, Object> argumentValueByActionArgumentNameMap) throws CommandException
+    {
+        Integer page = null;
+        final Object pageObject = argumentValueByActionArgumentNameMap.get("page");
+        if (null != pageObject)
+        {
+            page = (Integer) pageObject;
+        }
+        Colony colony = null;
+        final Object colonyObject = argumentValueByActionArgumentNameMap.get("colony");
+        if (null != colonyObject)
+        {
+            colony = (Colony) colonyObject;
+        }
+        executeShared(server, sender, colony, page);
+    }
+
     @Override
     public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
     {
         int colonyId = getIthArgument(args, 0, getColonyId(sender));
+        final Integer page = getIthArgument(args, 1, 1);
 
+        Colony colony = null;
         if (sender instanceof EntityPlayer)
         {
             if (colonyId == -1)
             {
-                final IColony colony = ColonyManager.getIColonyByOwner(sender.getEntityWorld(), (EntityPlayer) sender);
-                if (colony != null)
+                final IColony icolony = ColonyManager.getIColonyByOwner(sender.getEntityWorld(), (EntityPlayer) sender);
+                if (icolony != null)
                 {
-                    colonyId = colony.getID();
+                    colonyId = icolony.getID();
                 }
             }
+        }
+        colony = ColonyManager.getColony(colonyId);
 
+        executeShared(server, sender, colony, page);
+    }
+
+    private void executeShared(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, final Colony colony, Integer page) throws CommandException
+    {
+        if (null == page)
+        {
+            page = 1;
+        }
+
+        if (sender instanceof EntityPlayer)
+        {
             final EntityPlayer player = (EntityPlayer) sender;
-            if (!canPlayerUseCommand(player, LISTCITIZENS, colonyId))
+            if (null != colony)
             {
-                player.sendMessage(new TextComponentString("Not happenin bro!!, You are not permitted to do that!"));
-                return;
+                if (!canPlayerUseCommand(player, LISTCITIZENS, colony.getID()))
+                {
+                    player.sendMessage(new TextComponentString("Not happenin bro!!, You are not permitted to do that!"));
+                    return;
+                }
             }
         }
 
-        final Colony colony = ColonyManager.getColony(colonyId);
-
         final List<CitizenData> citizens = colony.getCitizenManager().getCitizens();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "colony" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2259)